CountingSort/app_resources/scatter_shader.comp.hlsl

```diff
@@ -14,16 +18,34 @@ uint32_t3 nbl::hlsl::glsl::gl_WorkGroupSize()
 void main(uint32_t3 ID : SV_GroupThreadID, uint32_t3 GroupID : SV_GroupID)
 {
     uint32_t tid = nbl::hlsl::workgroup::SubgroupContiguousIndex();
+
+    [unroll]
+    for (int i = 0; i < BucketsPerThread; i++)
+        sdata[BucketsPerThread * tid + i] = 0;
     uint32_t index = (nbl::hlsl::glsl::gl_WorkGroupID().x * WorkgroupSize + tid) * pushConstants.elementsPerWT;
 
     nbl::hlsl::glsl::barrier();
     
+    [unroll]
     for (int i = 0; i < pushConstants.elementsPerWT; i++)
     {
         if (index + i >= pushConstants.dataElementCount)
             break;
         uint32_t value = vk::RawBufferLoad(pushConstants.inputAddress + sizeof(uint32_t) * (index + i));
```

Comment:
learn about coalesced loads, when indexing elements to load you should have consecutive `InvocationIndex` load consecutive elements, your current sceheme is is really bad for cachelines

workgroup > subgroup > invocation

if you want to repeat elements, they should repeat at the workgroup granularity (i.e. isntead of invocation 0 processing element 0 and 1, they should process element 0, WorkgroupSize)

Reply:
Done